[gameplay]/killmessages/scripts/server/s.killmessages.lua

```diff
@@ -0,0 +1,164 @@
+        weapon = getPedWeapon(killer) or 51
+    end
+
+    if killer and (killerType == 'player') then
+        -- Suicide?
+        if (source == killer) then
+            if triggerEvent("onPlayerKillMessage", source, false, weapon, bodypart) then
+                eventTriggered(source, false, weapon, bodypart, usedVehicle)
+            end
+        else
+            if triggerEvent("onPlayerKillMessage", source, killer, weapon, bodypart) then
+                eventTriggered(source, killer, weapon, bodypart, usedVehicle)
+            end
+        end
+    else
+        -- Avoid showing deaths caused by explosions, cuz they are client-side
+        if (weapon ~= 63) then
+            if triggerEvent("onPlayerKillMessage", source, false, weapon, bodypart) then
+                eventTriggered(source, false, weapon, bodypart, usedVehicle)
+            end
+        end
+    end
+end
+addEventHandler('onPlayerWasted', root, createKillMessage)
+
+addEvent('outputKillFromClient', true)
+addEventHandler('outputKillFromClient', root, function(killer, veh)
+    if (source ~= client) then
+        return
+    end
+
+    if (not isElement(killer)) then
+        return
+    end
+
+    if (killer == source) then
+        if triggerEvent("onPlayerKillMessage", source, false, veh and 63 or 999) then
+            eventTriggered(source, false, veh and 63 or 999)
+        end
+    else
+        if triggerEvent("onPlayerKillMessage", client, killer, 19) then
+            eventTriggered(client, killer, 19, nil, veh)
+        end 
+    end
+end)
+
+function eventTriggered(player, killer, weapon, bodypart, vehicle)
+    outputConsoleMessage(player, killer, vehicle, weapon)
+
+    if isElement(vehicle) and (getElementType(vehicle) == 'vehicle') then
+        weapon = getElementModel(vehicle)
+    end
+
+    local pR, pG, pB = getPlayerColor(player)
+    local kR, kG, kB = getPlayerColor(killer)
+
+    outputKillMessage(player, pR, pG, pB, killer, kR, kG, kB, weapon)
+
+    return true
+end
+
+function outputKillMessage(player, pR, pG, pB, killer, kR, kG, kB, weapon)
+    if (not isElement(player)) or (getElementType(player) ~= 'player') then
+        outputDebugString("outputKillMessage - Invalid 'wasted' player specified", 0, 0, 0, 100)
+        return false
+    end
+
+    return triggerClientEvent(root, 'onClientPlayerKillMessage', player, killer, weapon, pR, pG, pB, kR, kG, kB)
+end
+
+function outputMessage(message, visibleTo, r, g, b, font)
+    if (type(message) ~= "string") and (type(message) ~= "table") then
+        outputDebugString("outputMessage - Bad 'message' argument", 0, 112, 112, 112)
+        return false
+    end
+    if (not isElement(visibleTo)) then
+        outputDebugString("outputMessage - Bad 'visibleTo' argument", 0, 112, 112, 112)
+        return false
+    end
+    return triggerClientEvent(visibleTo, "doOutputMessage", visibleTo, message, r, g, b, font)
```

Comment:
```suggestion
    return triggerClientEvent(visibleTo, "doOutputMessage", resourceRoot, message, r, g, b, font)
```
We need to fix this, since `visibleTo` can be a table.